v2/validator.go

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package v2
+
+import (
+	"fmt"
+	"github.com/go-playground/validator/v10"
+	"strings"
+)
+
+var val *validator.Validate
+
+func init() {
+	val = validator.New()
+}
+
+// Validate function will use the validator package to validate the struct annotation
+func Validate(a interface{}) error {
+	err := val.Struct(a)
+	// translate all error at once
+	if err != nil {
+		errs := err.(validator.ValidationErrors)
+		var errMsg []string
+		for _, e := range errs {
+			errMsg = append(errMsg, getErrorMessage(e))
+		}
+		return NewErrContractInvalid(strings.Join(errMsg, ", "))
```

Comment:
@lenny-intel may we use `\n` instead of comma to separate the error messages?  It would limit the error message cannot contain comma in the future.

Reply:
@cloudxxx8 , not wild about `\n` as it will impact how the message is logged. How about `; ` ?

Reply:
@lenny-intel sounds good to me.  @lindseysimple let's use `;`